### DISCUSSION TITLE
Add tracking code to <head>

### DISCUSSION THREAD

**Discussion by christiaanvaken** [YOURLS/YOURLS#3567] [Q&A]
I would like to add custom tracking code to <head> so that visits to a short url will be tracked. Where can I do this?

**Comment by dgw** [comment]
There is no `<head>` in a shortlink redirection. The client receives a 3xx HTTP status code and a `Location` header, then navigates to the value of `Location` without waiting for any HTML.

It's possible to call analytics endpoints server side (e.g. https://github.com/powerthazan/YOURLS-GA-MP-Tracking which would need updating to GA4), but as you didn't say what tracking solution you're using I can give any more specific pointers.

**Reply by christiaanvaken** [reply]
My mistake, I want to at Matomo (previous Piwik) for analytics tracking.

**Reply by dgw** [reply]
Do not know if it still works, but there is a Matomo plugin here: https://github.com/NothingTV/matomo-yourls

It's likely not the best method, as the author says in its readme. Could not readily find anything like the server side Google Analytics tracker for Matomo. 🙁

Probably better would be adding analytics campaign parameters to the long URL, assuming you are linking to properties you control.
